prism_compile.h

```diff
@@ -13,6 +13,11 @@ typedef struct pm_scope_node {
 
     ID *constants;
     st_table *index_lookup_table;
+
+    // Some locals are defined at higher scopes than they are used. We can use
+    // this offset to control which parent scopes local table we should be
+    // referencing from the current scope.
+    int local_depth_offset;
```

Comment:
Can we make this unsigned? Just to make it slightly clearer that this can never be `< 0`.

```suggestion
    unsigned int local_depth_offset;
```